File: oauth2-extended-authorization-server/authorization-server/src/main/java/com/onlyteo/sandbox/model/RegisterFormData.java
package com.onlyteo.sandbox.model;

import com.onlyteo.sandbox.validator.ValidPassword;
import jakarta.validation.constraints.Size;

@ValidPassword
public record RegisterFormData(
        @Size(min = 3, message = "{validation.username.Size.message}")
        String username,
        @Size(min = 6, message = "{validation.password.Size.message}")
        String password,
        String confirmPassword) {
}

File: spring-boot-oauth2-authorization-server-extended/spring-boot-oauth2-authorization-server-extended-server/src/main/kotlin/com/onlyteo/sandbox/model/RegisterFormData.kt
package com.onlyteo.sandbox.model

import com.onlyteo.sandbox.validator.ValidPassword
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

@ValidPassword
data class RegisterFormData(
    @field:Size(min = 3, message = "{validation.username.Size.message}") val username: String? = null,
    @field:Size(min = 6, message = "{validation.password.Size.message}") val password: String? = null,
    @field:NotBlank val confirmPassword: String? = null,
)
